catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaTopicPropertiesMetadata.java

```diff
@@ -6,12 +6,41 @@
 
 import com.datastrato.gravitino.connector.BasePropertiesMetadata;
 import com.datastrato.gravitino.connector.PropertyEntry;
-import java.util.Collections;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
+import java.util.List;
 import java.util.Map;
 
 public class KafkaTopicPropertiesMetadata extends BasePropertiesMetadata {
+  public static final String PARTITION_COUNT = "partition-count";
+  public static final String REPLICATION_FACTOR = "replication-factor";
+
+  private static final Map<String, PropertyEntry<?>> PROPERTIES_METADATA;
+
+  static {
+    List<PropertyEntry<?>> propertyEntries =
+        ImmutableList.of(
+            PropertyEntry.integerOptionalPropertyEntry(
+                PARTITION_COUNT,
+                "The number of partitions for the topic, if not specified, "
+                    + "will use the num.partition property in the broker",
+                true /* immutable */,
+                null /* default value */,
+                false /* hidden */),
+            // TODO: make REPLICATION_FACTOR mutable if needed
+            PropertyEntry.shortOptionalPropertyEntry(
+                REPLICATION_FACTOR,
+                "The number of replications for the topic, if not specified, "
+                    + "will use the default.replication.factor property in the broker",
+                true /* immutable */,
+                null /* default value */,
+                false /* hidden */));
+
+    PROPERTIES_METADATA = Maps.uniqueIndex(propertyEntries, PropertyEntry::getName);
+  }
+
```

Comment:
Suggested support for commonly used attributes.

Reply:
Could you please give some examples?

Reply:
At least，we'd better support the attributes : compression.type retention.ms min.insync.replicas